managers/asset-operations-manager.js

```diff
@@ -293,23 +306,90 @@ export default class AssetOperationsManager {
             minimumNumberOfNodeReplications,
         );
 
-        let dataset;
-
+        let dataset = {};
         if (typeof content === 'string') {
-            dataset = content
-                .split('\n')
-                .map((line) => line.trimStart().trimEnd())
-                .filter((line) => line.trim() !== '');
+            dataset.public = this.processContent(content);
+        } else if (
+            typeof content.public === 'string' ||
+            (!content.public && content.private && typeof content.private === 'string')
+        ) {
+            if (content.public) {
+                dataset.public = this.processContent(content.public);
+            }
+            if (content.private && typeof content.private === 'string') {
+                dataset.private = this.processContent(content.private);
+            }
         } else {
             dataset = await kcTools.formatDataset(content);
         }
 
-        const numberOfChunks = kcTools.calculateNumberOfChunks(dataset, CHUNK_BYTE_SIZE);
+        dataset.public = kcTools.generateMissingIdsForBlankNodes(dataset.public);
+        if (dataset.private) {
+            dataset.private = kcTools.generateMissingIdsForBlankNodes(dataset.private);
+        }
+
+        let privateTripletsSortedAndGrouped;
```

Comment:
i think triple is triplet on english, triplets == triples so i would change it like that